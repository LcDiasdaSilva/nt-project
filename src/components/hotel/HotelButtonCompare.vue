<template>
  <div style="text-align: right">
    <button
      :disabled="total < 2"
      class="customHotelCompareButton"
      type="button"
      @click="submit"
    >
      <label>Comparar</label>
      <label v-if="total > 1">
        <b class="mx-2">{{ total }}</b> hot√©is</label
      >
    </button>
    <small>* Seleciona 2 ou mais hotel para comparar</small>
  </div>
</template>
<script setup lang="ts">
  import { useHotelsCompareStore } from "../../store/useHotelCompareStore";
  import { computed, defineEmits } from "vue";
  const hotelCompareStore = useHotelsCompareStore();

  const emit = defineEmits(["submit"]);

  const total = computed(() => {
    return hotelCompareStore.getStateHotelsCompare.length;
  });

  function submit() {
    emit("submit");
  }
</script>
