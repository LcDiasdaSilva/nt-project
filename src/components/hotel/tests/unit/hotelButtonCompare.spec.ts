import { shallowMount } from "@vue/test-utils";
import HotelButtonCompare from "../../HotelButtonCompare.vue";
import { createTestingPinia } from "@pinia/testing";


describe("HotelButtonCompare.vue", () => {
  it("matches snapshot when total hotels to compare is less than 2", () => {
    const wrapper = shallowMount(HotelButtonCompare, {
      global: {
        plugins: [createTestingPinia({ 
          initialState: {
            hotelCompareStore: {
              getStateHotelsCompare: []
            }
          }
        })],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot when total hotels to compare is 2 or more", () => {
    const wrapper = shallowMount(HotelButtonCompare, {
      global: {
        plugins: [createTestingPinia({
          initialState: {
            hotelCompareStore: {
              getStateHotelsCompare: [{}, {}]
            }
          }
        })],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
