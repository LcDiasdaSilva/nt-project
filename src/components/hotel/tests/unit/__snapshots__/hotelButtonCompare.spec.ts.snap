// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelButtonCompare.vue matches snapshot when total hotels to compare is 2 or more 1`] = `
<div style="text-align: right;"><button disabled="" class="customHotelCompareButton" type="button"><label>Comparar</label>
    <!--v-if-->
  </button><small>* Selecione até 3 hotéis para comparar</small></div>
`;

exports[`HotelButtonCompare.vue matches snapshot when total hotels to compare is less than 2 1`] = `
<div style="text-align: right;"><button disabled="" class="customHotelCompareButton" type="button"><label>Comparar</label>
    <!--v-if-->
  </button><small>* Selecione até 3 hotéis para comparar</small></div>
`;
